### PR TITLE
Refactor database drivers - push up desctructor

### PR DESCRIPTION
**Pull request by izharaazmi**
#### Summary of Changes
1. Push up the destructor from various database drivers to their parent abstract class JDatabaseDriver as it almost every time just does one thing viz `disconnect()`. Otherwise it can always be overridden.
2. Never `unset` class property. `$this->connection` set to null in disconnect methods of the db drivers.
3. Nosql connect return type is void not bool.
#### Testing Instructions

Make sure the disconnect / connect using all the database drivers work correctly without any warning/notice.

Without this patch calling disconnect followed by connect will raise PHP warning. After this patch it should work fine.

### CODE REVIEW

**Comment by izharaazmi** [comment]
Any review on this please?

File: libraries/joomla/database/driver.php [hunk 1]
```diff
@@ -670,6 +670,16 @@ public function createDatabase($options, $utf = true)
 	}
 
 	/**
+	 * Destructor.
+	 *
+	 * @since   CMS 3.6
```
Comment (zero-24) [inline libraries/joomla/database/driver.php]: please use `__DEPLOY_VERSION__` without the CMS Prefix ;)

Comment (izharaazmi) [inline libraries/joomla/database/driver.php]: Sure, I noticed this. Probably those days this macro wasn't born yet ;-) 

I'll fix it when on desk.

Comment (zero-24) [inline libraries/joomla/database/driver.php]: That is correct ;)

**Review by mbabker — APPROVED** [review]

**Review by rdeutz — APPROVED** [review]

**Pull request merged**